generators/new_collateral/config/parser_test.go

```diff
@@ -0,0 +1,53 @@
+package config_test
+
+import (
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/config"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/test_data"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Config Parser", func() {
+	var (
+		testConfigFilePath = "../test_data/"
+		testConfigFileName = "testConfig"
+		configParser       = config.Parser{}
+	)
+
+	Context("ParseCurrentConfigFile", func() {
+		It("reads in the exporter metadata", func() {
+			expectedExporterMetadata := types.ExporterMetaData{
+				Home:             "github.com/makerdao/vulcanizedb",
+				Name:             "transformerExporter",
+				Save:             false,
+				Schema:           "maker",
+				TransformerNames: []string{"cat_v1_1_0", "cat_file_vow"},
+			}
+
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, testConfigFileName)
+			Expect(parseErr).NotTo(HaveOccurred())
+			Expect(config.ExporterMetadata).To(Equal(expectedExporterMetadata))
+		})
+
+		It("reads in the exporterTransformers", func() {
+			expectedTransformerExporters := types.TransformerExporters{
+				"cat_v1_1_0":   test_data.Cat110Exporter,
+				"cat_file_vow": test_data.CatFileVowExporter,
+			}
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, testConfigFileName)
+			Expect(parseErr).NotTo(HaveOccurred())
+			Expect(config.TransformerExporters).To(Equal(expectedTransformerExporters))
+		})
+
+		It("reads in the contracts", func() {
+			expectedContracts := types.Contracts{
+				"MCD_CAT_1_0_0": test_data.Cat100Contract,
+				"MCD_CAT_1_1_0": test_data.Cat110Contract,
+			}
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, testConfigFileName)
+			Expect(parseErr).NotTo(HaveOccurred())
+			Expect(config.Contracts).To(Equal(expectedContracts))
+		})
+	})
```

Comment:
What about the expected errors?

Reply:
good call! added tests in 938de7a859b2a550b95a0efe656128c314b15f6e

I'd be interested in your thoughts about other ways to test some of the functionality in the parser - the easiest way I could think of was to export some of the previously private methods, but not sure if maybe there's a better way to handle this that I'm not seeing at the moment.